### DISCUSSION TITLE
INFO Start step 200 'Generate supply plan' disconnect when run over 1 HR

### DISCUSSION THREAD

**Discussion by maxware0715** [frePPLe/frepple#644] [Q&A] [answered]
Our Frepple version is 6.25

As mentioned in https://github.com/frePPLe/frepple/discussions/643
I dig into log and find disconnection always happened in task 'Generate supply plan':

INFO Start step 200 'Generate supply plan' at 14:45:50
INFO Finished 'Generate supply plan' at 15:55:34

When this task exceed 1 hr, it will surely trigger following error:
Error during planning: connection already closed
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.OperationalError: server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.

This error shows in log file, while Task Status in Frepple UI don't show Failed and remain running state.

Guess there has some setting cut off the task right after 1 hour.
All planning job run step 200 less than this duration end successfully.
Any suggestions?

**Comment by jdetaeye** [comment]
This looks like postgresql cutting the connection.   There are settings on postgres to restrict excessiblely long queries or connections - eg statement_timeout 
You might check the postgres logs to find a smoking gun.

**Comment by maxware0715** [comment, marked as answer]
I solve this issue by manually close old connection:

freppledb\common\commands.py
class PlanTaskSequence(PlanTask): 
  def run():
    from django.db import close_old_connections              
    close_old_connections()

but root cause still not found.
guess caused by either psycopg2 or my infra environment settings
